modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/core/DefaultRaftClientService.java

```diff
@@ -88,19 +93,78 @@ public Future<Message> requestVote(final PeerId peerId, final RequestVoteRequest
 
     @Override
     public Future<Message> appendEntries(final PeerId peerId, final AppendEntriesRequest request,
-        final int timeoutMs, final RpcResponseClosure<AppendEntriesResponse> done) {
+            final int timeoutMs, final RpcResponseClosure<AppendEntriesResponse> done) {
 
         // Assign an executor in round-robin fasion.
         final Executor executor = this.appendEntriesExecutorMap.computeIfAbsent(peerId,
-            k -> nodeOptions.getStripedExecutor().next());
+                k -> nodeOptions.getStripedExecutor().next());
 
         if (connect(peerId)) { // Replicator should be started asynchronously by node joined event.
+            if (isHeartbeatRequest(request)) {
```

Comment:
This check is already done in the upper level in
org.apache.ignite.raft.jraft.core.Replicator#sendEmptyEntries(boolean, org.apache.ignite.raft.jraft.rpc.RpcResponseClosure<org.apache.ignite.raft.jraft.rpc.RpcRequests.AppendEntriesResponse>, boolean)
Just call sendHeartbeat from this method.
```
           if (isHeartbeat) {
                ....
                this.heartbeatInFly = this.rpcService.sendHeartbeat(this.options.getPeerId().getEndpoint(), request, coalesce,
                        this.options.getElectionTimeoutMs() / 2, heartbeatDone);
            }
```

Reply:
It was implemented this way in the PoC without any dependency issues.
Ok. let's keep it as is, but at least remove copypaste by moving isHearbeat(..) to utitlities class.

Reply:
Done.